### ISSUE TITLE
Memory Management

### ISSUE THREAD

**Issue by patrickbst** [A2osX/A2osX#59]
Loading the network via script versus at the command line consumes different amounts of   memory.  In addition, killing some processes such as networkd do not release all of their memory back.  Added related problem, if DHCP lease fails and you do IPCONFIG -D it consumes even more memory.

From XLS 190

**Comment by patrickbst** [comment]
This issue may no longer be valid as the commands for starting the network have changed.  Is it now possible to unload the network library?   Can you unload the hardware driver after unloading the net lib?  Changing to question for dev; will close after response and update of doc if needed.

**Comment by patrickbst** [comment]
I did some quick tests with Build 1703 and loading network in INIT, with a script (NET), or on the command line manually.  Booting /FULLBOOT (STABLE) unchanged MEM reports 21,985 free.  If you load U2/NETWORKD in INIT, mem is 14049.  Via a script, mem is 13105 and manually it is 13825.  Clearly the preferred option is via INIT.  I am guessing the difference between these scenarios is mem holes that can get created from loading/running scripts.  Will note this in doc.

Still need answer about being able to unload NETWORKD and U2.drv.

**Comment by patrickbst** [comment]
Notes from development about this to update doc and close:
about unloading daemon
networkd will load libtcpip, then call ipconfig to read etc/hosts.....and eventually get a dhcp lease
networkd stays in mem with ONE lock on libtcpip
ipconfig, while doing its job got another lock on libtcpip -> locks=2......then ipconfig exits -> locks= 1
then you load TELNETD
lock = 2 on libtcpip
networkd is "annimating" all the net stack polling ETHx.....passing frames to protocols (tcpip, ETALK....) sending periodic time event to expire cache, retransmit frames...etc....
if you KILL TELNETD ......-> lock = 1
if you kill NETWORKD -> lock = 0.....so the libtcpip will unload
if you kill NETWORKD first......TELNETD will stay in a strange state (no more pulse from NETWORKD) and will lock libtcpip forever....
so basically, you could unload network, by first killing IP based apps, then finally NETWORKD
it should bring libtcpip to 3 2 1 ......unload
there is no way to unload DEV because the mmmgr dealing with this KERNEL zone is very primitive and don't allow fragmentation
so just use normal KILL to kill things in reverse.  but also part of what i was looking for, was could you write a script that load drv and then netd, telnet (to a80s) then when telnet exit, could jill networkd  and get most of mem back. 
just, make sure all connection are closed before  unloading network

**Comment by patrickbst** [comment]
Notes on this have been added to the User Guide.  Closing.
